mbed-client-libservice/nsdynmemLIB.h

```diff
@@ -67,6 +67,16 @@ typedef struct mem_stat_t {
     uint32_t heap_alloc_fail_cnt;               /**< Counter for Heap allocation fail. */
 } mem_stat_t;
 
+/**
+ * /struct mem_gen_stat_t
```

Comment:
Is this just extra complexity why was the previous stat enough? it does not look like you are saving much

Reply:
You forget that user must enable those previous stats. This change enable possibility to get minimal information always at any heap user. Old system works when app enable heap.

Reply:
and that old stat structure is giving by user.

Reply:
I did not forget that was the point why not always enable it. because this just brings complexity without saving anything

Reply:
So my suggestion is that by default always enable stats and if user gives it then use the structure from the user and not the default internal

Reply:
That is then API change. Now user reset some of the values at structure I believe.  Actually simulator port use own structure if user not gives so it works then Like you want. const pointer can't edit any values. But then it actually works like we want. I start like your idea. Lets make that like at simulator platform.